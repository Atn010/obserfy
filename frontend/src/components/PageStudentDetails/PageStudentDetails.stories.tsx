import React, { FC } from "react"
import PageStudentDetails from "./PageStudentDetails"

export default {
  title: "Page|PageStudentDetails",
  component: PageStudentDetails,
  parameters: {
    componentSubtitle: "Just a simple PageStudentDetails",
  },
}

export const Basic: FC = () => <PageStudentDetails id="asdfadsf" />
