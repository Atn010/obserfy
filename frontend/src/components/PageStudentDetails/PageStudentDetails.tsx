import React, { FC, useState } from "react"
import { navigate } from "gatsby"
import { useQueryStudentDetails } from "../../hooks/students/useQueryStudentDetails"
import Card from "../Card/Card"
import Flex from "../Flex/Flex"
import Box from "../Box/Box"
import Typography from "../Typography/Typography"
import Spacer from "../Spacer/Spacer"
import Icon from "../Icon/Icon"
import { ReactComponent as NextIcon } from "../../icons/next-arrow.svg"
import EmptyListPlaceholder from "../EmptyListPlaceholder/EmptyListPlaceholder"
import AddObservationDialog from "../AddObservationDialog/AddObservationDialog"
import MockAvatar from "../mockAvatar"
import Button from "../Button/Button"
import FloatingActionButton from "../FloatingActionButton/FloatingActionButton"
import { ReactComponent as PlusIcon } from "../../icons/plus.svg"

export interface Observation {
  shortDesc: string
  longDesc: string
}
interface Props {
  id: string
}
export const PageStudentDetails: FC<Props> = ({ id }) => {
  const [showAddObservationDialog, setShowObservationDialog] = useState(false)
  const [observations, setObservations] = useState<Observation[]>([])
  const [editObservations, setEditObservations] = useState()
  const [details] = useQueryStudentDetails(id)

  function addObservation(): void {
    setEditObservations(undefined)
    setShowObservationDialog(true)
  }

  const listOfObservations = observations.map(({ longDesc, shortDesc }) => (
    <Card
      p={3}
      mb={3}
      onClick={() => {
        setEditObservations({ longDesc, shortDesc })
        setShowObservationDialog(true)
      }}
    >
      <Flex alignItems="center">
        <Box>
          <Typography.Body fontSize={1} color="textMediumEmphasis">
            Thursday, 26 Dec &amp;19
          </Typography.Body>
          <Typography.H5>{shortDesc}</Typography.H5>
        </Box>
        <Spacer />
        <Icon as={NextIcon} m={0} />
      </Flex>
    </Card>
  ))

  const emptyObservationPlaceholder = observations.length === 0 && (
    <EmptyListPlaceholder
      text="What did you observe?"
      callToActionText="add observation"
      onActionClick={addObservation}
    />
  )

  const addObservationDialog = showAddObservationDialog && (
    <AddObservationDialog
      defaultValue={editObservations}
      onCancel={() => setShowObservationDialog(false)}
      onConfirm={name => {
        setObservations([...observations, name])
        setShowObservationDialog(false)
      }}
    />
  )

  return (
    <>
      <Box>
        <Flex alignItems="center" p={3}>
          <MockAvatar size={60} />
          <Typography.H3>{details?.name}</Typography.H3>
        </Flex>
        <Button
          ml={3}
          variant="outline"
          onClick={() => navigate(`/students/edit?id=${id}`)}
        >
          Edit student
        </Button>
        <Box p={3}>
          <Typography.H4 color="textMediumEmphasis" mb={2}>
            Observations
          </Typography.H4>
          {emptyObservationPlaceholder}
          {listOfObservations}
        </Box>
      </Box>
      <FloatingActionButton onClick={addObservation}>
        <Icon as={PlusIcon} m={0} mr={2} fill="onPrimary" /> New observation
      </FloatingActionButton>
      {addObservationDialog}
    </>
  )
}

export default PageStudentDetails
