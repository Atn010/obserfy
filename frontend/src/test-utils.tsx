/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { ThemeProvider } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { IntlProvider } from "gatsby-plugin-intl3"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import Theme from "./gatsby-plugin-theme-ui"
import Box from "./components/Box/Box"
import en from "./intl/en.json"

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <LocationProvider history={history}>
      <ThemeProvider theme={Theme}>
        <IntlProvider locale="en" messages={en}>
          <Box fontSize={[16, 20]}>{children}</Box>
        </IntlProvider>
      </ThemeProvider>
    </LocationProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
