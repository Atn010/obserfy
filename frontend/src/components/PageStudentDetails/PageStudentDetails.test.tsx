import React from "react"
import { render } from "../../test-utils"
import { Basic } from "./PageStudentDetails.stories"

describe("PageStudentDetails", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
