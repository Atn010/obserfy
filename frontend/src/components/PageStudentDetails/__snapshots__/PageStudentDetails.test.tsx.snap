// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageStudentDetails should render correctly 1`] = `
.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width:40em) {
  .emotion-14 {
    font-size: 20px;
  }
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  object-fit: cover;
  object-position: center;
  border-radius: 99999px;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  border-width: 1px;
  border-style: solid;
  overflow: hidden;
  width: 60px;
  height: 60px;
  margin-right: 16px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 24px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 300;
  line-height: heading;
  font-size: 3.90625rem;
  line-height: 6rem;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary,#6E55C6);
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  text-transform: capitalize;
  box-sizing: border-box;
  font-size: 14px;
  font-weight: bold;
  color: var(--theme-ui-colors-textPrimary,#3A2B95);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  border: 1px solid;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 16px;
}

.emotion-3:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-3:hover,
.emotion-3:focus {
  background-color: var(--theme-ui-colors-primaryLighter,rgba(161,130,250,0.12));
  border-color: var(--theme-ui-colors-primary,#6E55C6);
  outline: none;
}

.emotion-3:hover:disabled,
.emotion-3:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 24px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 300;
  line-height: heading;
  font-size: 3.125rem;
  line-height: 6rem;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: var(--theme-ui-colors-textMediumEmphasis,rgba(0,0,0,0.60));
  margin-bottom: 8px;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 8px;
  background-color: var(--theme-ui-colors-background,#F7F8FA);
  border-radius: 6px;
  box-shadow: rgba(0,0,0,0.1) 0px 0px 1px,rgba(0,0,0,0.27) 0px 2px 4px -2px;
  padding: 0;
  background-color: var(--theme-ui-colors-surface,#fff);
  overflow: hidden;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 16px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 24px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 300;
  line-height: heading;
  font-size: 2rem;
  line-height: 3rem;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  margin-bottom: 16px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary,#6E55C6);
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  text-transform: capitalize;
  box-sizing: border-box;
  font-size: 14px;
  font-weight: bold;
  color: var(--theme-ui-colors-textPrimary,#3A2B95);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  border: 1px solid;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-7:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-7:hover,
.emotion-7:focus {
  background-color: var(--theme-ui-colors-primaryLighter,rgba(161,130,250,0.12));
  border-color: var(--theme-ui-colors-primary,#6E55C6);
  outline: none;
}

.emotion-7:hover:disabled,
.emotion-7:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border,rgba(0,0,0,.1));
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon,rgba(0,0,0,0.6));
  margin: 0;
  height: 20px;
  width: 20px;
  margin-right: 8px;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary,#6E55C6);
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  text-transform: capitalize;
  box-sizing: border-box;
  font-size: 14px;
  font-weight: bold;
  color: var(--theme-ui-colors-onPrimary,#fff);
  background-color: var(--theme-ui-colors-primary,#6E55C6);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  transition: background-color 125ms cubic-bezier(0.0,0.0,0.2,1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  box-shadow: rgba(0,0,0,0.1) 0px 0px 1px,rgba(0,0,0,0.27) 0px 2px 4px -2px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 99999px;
  position: fixed;
  bottom: 0;
  right: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 50px;
  min-height: 50px;
  margin: 16px;
}

.emotion-13:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-13:hover,
.emotion-13:focus {
  background-color: var(--theme-ui-colors-primaryDark,#3A2B95);
  outline: none;
}

.emotion-13:hover:disabled,
.emotion-13:focus:disabled {
  background-color: var(--theme-ui-colors-primary,#6E55C6);
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-onPrimary,#fff);
  margin: 0;
  height: 20px;
  width: 20px;
  margin-right: 8px;
}

<div>
  <div
    class="emotion-14"
  >
    <div
      class="emotion-11"
    >
      <div
        class="emotion-2"
      >
        <img
          class="emotion-0"
          src="https://images.unsplash.com/photo-1571942727532-a67a4bf9845a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1000&q=80"
        />
        <h3
          class="emotion-1"
        />
      </div>
      <button
        class="emotion-3"
      >
        Edit student
      </button>
      <div
        class="emotion-10"
      >
        <h4
          class="emotion-4"
        >
          Observations
        </h4>
        <div
          class="emotion-9"
        >
          <div
            class="emotion-8"
          >
            <h6
              class="emotion-5"
            >
              What did you observe?
            </h6>
            <button
              class="emotion-7"
            >
              <div
                class="emotion-6"
              />
              add observation
            </button>
          </div>
        </div>
      </div>
    </div>
    <button
      class="emotion-13"
    >
      <div
        class="emotion-12"
      />
       New observation
    </button>
  </div>
</div>
`;
